docs/conceptual/core/executor.md

```diff
@@ -2,8 +2,151 @@
+Substrate runtime.
 
-* Native VS Wasm Execution
+## Runtime Execution
 
-* Forkless Runtime Upgrades
\ No newline at end of file
+The Substrate runtime is compiled into a native executable and a WebAssembly
+(Wasm) binary.
+
+The native runtime is included as part of the node executable, while the Wasm
+binary is stored on the blockchain under a well known storage key.
+
+These two representations of the runtime may not be the same. For example: when
+the runtime is upgraded. The executor determines which version of the runtime to
+use when dispatching and executing calls.
+
+### Execution Strategy
+
+Before runtime execution begins, the Substrate client proposes which runtime
+execution environment should be used. This is controlled by the execution
+strategy, which can be configured for the different parts of the blockchain
+execution process. The strategies are:
+
+- `Native`: Execute with native build (if available, WebAssembly otherwise); if
+  it fails, do not fallback to Wasm.
+- `Wasm`: Only execute with the WebAssembly build.
+- `Both`: Execute with both native (where available) and WebAssembly builds.
+- `NativeElseWasm`: Execute with the native build if possible; if it fails, then
+  execute with WebAssembly.
+
+The default execution strategies for the different parts of the blockchain
+execution process are:
+
+- Syncing: `NativeElseWasm`
+- Block Import: `NativeElseWasm`
+- Block Construction: `Wasm`
+- Off-Chain Worker: `Native`
+- Other: `Native`
+
+### Wasm Execution
+
+The Wasm representation of the Substrate runtime is considered the canonical
+runtime. Because this Wasm runtime is placed in the blockchain storage, the
+network must come to consensus about this binary. Thus it can be verified to be
+consistent across all syncing nodes.
+
+The Wasm execution environment can be more restrictive than the native execution
+environment. For example, the Wasm runtime always executes in a 32-bit
+environment with a 4 GB memory limit.
+
+For these reasons, the blockchain prefers to do block construction with the Wasm
+runtime even though Wasm execution is measurably slower than native execution.
+Some logic executed in Wasm will always work in the native execution
+environment, but the same cannot be said the other way around. Wasm execution
+can help to ensure that block producers create valid blocks.
+
+### Native Execution
+
+The native runtime will only be used by the executor when it is chosen as the
+execution strategy and it is compatible with the requested runtime version. For
+all other execution processes other than block construction, the native runtime
+is preferred since it is more performant. In any situation where the native
+executable should not be run, the canonical Wasm runtime is executed instead.
+
+## Runtime Upgrades
+
+The Substrate runtime is designed to be upgraded so that you can continue to
+iterate on your state transition function even after your blockchain goes live.
```

Comment:
"iterate on your state transition function"? I was thinking of either swapping `iterate on` for `adjust` or making this change (and the above one)
```suggestion
upgrade your state transition function even after your blockchain goes live.
```

Reply:
I don't like using upgrade twice in a sentence. Feels weird right?

I would be fine with `adjust` but would want to understand the change.

Reply:
with the change below it seems fine

Reply:
I *feel* like iterate is the wrong word here. [from mwebster](https://www.merriam-webster.com/dictionary/iteration): a procedure in which repetition of a sequence of operations yields results successively closer to a desired result. Maybe it does work...

That's why I'm more in favor of changing the position of "upgrade", but idrc...